### PR TITLE
update bcj threshold

### PR DESCRIPTION
**Pull request by hanfei1991**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -328,16 +328,16 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 - 作用域：SESSION | GLOBAL
 - 默认值：`10240`
-- 范围：`[0, 9223372036854775807]`
-- 单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 范围：`[-1, 9223372036854775807]`
+- 单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当值为 -1 时，默认选择 broadcast_join 算法。
```
Comment (LittleFall) [inline system-variables.md:332]: add a ref to https://docs.pingcap.com/zh/tidb/dev/use-tiflash#mpp-%E6%A8%A1%E5%BC%8F%E7%9A%84%E7%AE%97%E6%B3%95%E6%94%AF%E6%8C%81?

File: system-variables.md [hunk 2]
```diff
@@ -328,16 +328,16 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 
 ### `tidb_broadcast_join_threshold_size` <span class="version-mark">从 v5.0 版本开始引入</span>
 
 - 作用域：SESSION | GLOBAL
 - 默认值：`104857600` (100 MiB)
-- 范围：`[0, 9223372036854775807]`
+- 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当值为 -1 时，默认选择 broadcast_join 算法。
```
Comment (LittleFall) [inline system-variables.md:340]: ditto

Comment (LittleFall) [reply system-variables.md:340]: add tidb_opt_broadcast_cartesian_join

**Review by LittleFall — APPROVED** [review]
LGTM

**Review by LiSong0214 — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -328,16 +328,16 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 - 作用域：SESSION | GLOBAL
 - 默认值：`10240`
-- 范围：`[0, 9223372036854775807]`
-- 单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 范围：`[-1, 9223372036854775807]`
+- 单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，代表无穷大。
```
Comment (TomShawn) [inline system-variables.md:332]:
```suggestion
- 单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，该阈值无穷大。
```

File: system-variables.md [hunk 4]
```diff
@@ -328,16 +328,16 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
 
 ### `tidb_broadcast_join_threshold_size` <span class="version-mark">从 v5.0 版本开始引入</span>
 
 - 作用域：SESSION | GLOBAL
 - 默认值：`104857600` (100 MiB)
-- 范围：`[0, 9223372036854775807]`
+- 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，代表无穷大。
```
Comment (TomShawn) [inline system-variables.md:340]:
```suggestion
- 如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，该阈值无穷大。
```

File: tiflash/use-tiflash.md [hunk 5]
```diff
@@ -347,8 +347,8 @@ mysql> explain select count(*) from customer c join nation n on c.c_nationkey=n.
 
 TiFlash 提供了两个全局/会话变量决定是否选择 Broadcast Hash Join，分别为：
 
-- [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
-- [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。
+- [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，代表无穷大。
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:350]:
```suggestion
- [`tidb_broadcast_join_threshold_size`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为 bytes。如果表大小（字节数）小于该值，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，该阈值无穷大。
```

File: tiflash/use-tiflash.md [hunk 6]
```diff
@@ -347,8 +347,8 @@ mysql> explain select count(*) from customer c join nation n on c.c_nationkey=n.
...
+- [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，代表无穷大。
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:351]:
```suggestion
- [`tidb_broadcast_join_threshold_count`](/system-variables.md#tidb_broadcast_join_threshold_count-从-v50-版本开始引入)，单位为行数。如果 join 的对象为子查询，优化器无法估计子查询结果集大小，在这种情况下通过结果集行数判断。如果子查询的行数估计值小于该变量，则选择 Broadcast Hash Join 算法。否则选择 Shuffled Hash Join 算法。当设置为 -1 时，该阈值无穷大。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**